.github/workflows/cicd.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This is the top-level workflow that runs on each pull request and push to main.
+# It invokes other jobs to perform builds and run tests.
+# All jobs run in parallel, using build artifacts to synchronize jobs.
+
+name: CI/CD
+
+# Run on push so we can capture the baseline code coverage for any squash commit.
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+permissions:
+  contents: read
+  checks: read  # Required by reusable-test.yml to check build status.
+
+jobs:
+  # Perform the regular build.
+  regular:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64
+
+  # Build with C++ static analyzer.
+  analyze:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64-Analyze
+      build_options: /p:Analysis='True'
+
+  # Build with C++ address sanitizer.
+  sanitize:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64-Sanitize
+      build_options: /p:Sanitizer='True'
+
+  # Run the unit tests in GitHub.
+  unit_tests:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      name: unit_tests
+      test_command: unit_tests.exe -s
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: windows-2019
+      code_coverage: true
+      gather_dumps: true
+
+  # Run the API tests in GitHub.
+  ebpf_client:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      name: ebpf_client
+      test_command: ebpf_client.exe -s
```

Comment:
remove this. Redundant as it runs in selfhost runner. In github runner it wont exercise the kernel code path. The mock code path is executed by unit test anyways.

Reply:
Sure, happy to remove this after we add the code coverage driver test run.